esphome/components/bl0906/sensor.py

```diff
@@ -0,0 +1,176 @@
+                state_class=STATE_CLASS_TOTAL,
+                unit_of_measurement=UNIT_KILOWATT_HOURS,
+            ),
+        }
+    )
+    .extend(
+        cv.Schema(
+            {
+                cv.Optional(f"{CONF_CHANNEL}_{i+1}"): cv.Schema(
+                    {
+                        cv.Optional(CONF_CURRENT): cv.maybe_simple_value(
+                            sensor.sensor_schema(
+                                icon=ICON_CURRENT_AC,
+                                accuracy_decimals=3,
+                                device_class=DEVICE_CLASS_CURRENT,
+                                unit_of_measurement=UNIT_AMPERE,
+                            ),
+                            key=CONF_NAME,
+                        ),
+                        cv.Optional(CONF_POWER): cv.maybe_simple_value(
+                            sensor.sensor_schema(
+                                icon=ICON_POWER,
+                                accuracy_decimals=0,
+                                device_class=DEVICE_CLASS_POWER,
+                                unit_of_measurement=UNIT_WATT,
+                            ),
+                            key=CONF_NAME,
+                        ),
+                        cv.Optional(CONF_ENERGY): cv.maybe_simple_value(
+                            sensor.sensor_schema(
+                                icon=ICON_ENERGY,
+                                accuracy_decimals=3,
+                                device_class=DEVICE_CLASS_ENERGY,
+                                unit_of_measurement=UNIT_KILOWATT_HOURS,
+                            ),
+                            key=CONF_NAME,
+                        ),
+                    }
+                )
+                for i in range(6)
+            }
+        )
+    )
+    .extend(uart.UART_DEVICE_SCHEMA)
+    .extend(cv.polling_component_schema("60s"))
+)
+
+
+@automation.register_action(
+    "bl0906.reset_energy",
+    ResetEnergyAction,
+    maybe_simple_id(
+        {
+            cv.Required(CONF_ID): cv.use_id(BL0906),
+        }
+    ),
+)
+async def reset_energy_to_code(config, action_id, template_arg, args):
+    paren = await cg.get_variable(config[CONF_ID])
+    return cg.new_Pvariable(action_id, template_arg, paren)
+
+
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await uart.register_uart_device(var, config)
+    if frequency_config := config.get(CONF_FREQUENCY):
+        sens = await sensor.new_sensor(frequency_config)
+        cg.add(var.set_frequency_sensor(sens))
+    if temperature_config := config.get(CONF_TEMPERATURE):
+        sens = await sensor.new_sensor(temperature_config)
+        cg.add(var.set_temperature_sensor(sens))
+    if voltage_config := config.get(CONF_VOLTAGE):
+        sens = await sensor.new_sensor(voltage_config)
+        cg.add(var.set_voltage_sensor(sens))
+
+    for i in range(8):
```

Comment:
I think I accidentally put 8 instead of 6 here
```suggestion
    for i in range(6):
```